internal/operator/internal/operator_installer.go

```diff
@@ -62,3 +66,67 @@ func (o OperatorInstaller) InstallOperator(ctx context.Context) (*v1alpha1.Clust
 
 	return todo, nil
 }
+
+func (o OperatorInstaller) verifyOperatorInstallation(ctx context.Context, ip *v1alpha1.InstallPlan) (*v1alpha1.ClusterServiceVersion, error) {
+	ipKey := types.NamespacedName{
+		Namespace: ip.GetNamespace(),
+		Name:      ip.GetName(),
+	}
+	if err := wait.PollImmediateUntil(time.Millisecond*250, func() (bool, error) {
+		if err := o.cfg.Client.Get(ctx, ipKey, ip); err != nil {
+			return false, err
+		}
+		if ip.Status.Phase == v1alpha1.InstallPlanPhaseComplete {
+			return true, nil
+		}
+		return false, nil
+	}, ctx.Done()); err != nil {
+		return nil, fmt.Errorf("waiting for operator installation to complete: %v", err)
+	}
+
+	// verify CSV
+	csv, err := o.verifyCSV(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("waiting for CSV installation to complete: %v", err)
+	}
+	return csv, nil
+}
+
+func (o OperatorInstaller) verifyCSV(ctx context.Context) (*v1alpha1.ClusterServiceVersion, error) {
+
+	csvKey := types.NamespacedName{
+		Namespace: o.cfg.Namespace,
+	}
+
+	if csvKey.Name == "" {
+		return nil, fmt.Errorf("could not find installed CSV in install plan")
+	}
```

Comment:
This will _always_ result in the error being returned right? Should we pass the install plan to this function so we can find the CSV key?

Reply:
Moved csvKey login into install Plan function.